File: aspnetcore/src/IO.Swagger/Models/V3StoppingPatternStop.cs
/*
 * PTV Timetable API - Version 3
 *
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * OpenAPI spec version: v3
 * 
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 */
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Models
{
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class V3StoppingPatternStop : IEquatable<V3StoppingPatternStop>
    { 
        /// <summary>
        /// Gets or Sets StopTicket
        /// </summary>

        [DataMember(Name="stop_ticket")]
        public V3StopTicket StopTicket { get; set; }

        /// <summary>
        /// Distance of stop from input location (in metres); returns 0 if no location is input
        /// </summary>
        /// <value>Distance of stop from input location (in metres); returns 0 if no location is input</value>

        [DataMember(Name="stop_distance")]
        public float? StopDistance { get; set; }

        /// <summary>
        /// suburb of stop
        /// </summary>
        /// <value>suburb of stop</value>

        [DataMember(Name="stop_suburb")]
        public string StopSuburb { get; set; }

        /// <summary>
        /// Name of stop
        /// </summary>
        /// <value>Name of stop</value>

        [DataMember(Name="stop_name")]
        public string StopName { get; set; }

        /// <summary>
        /// Stop identifier
        /// </summary>
        /// <value>Stop identifier</value>

        [DataMember(Name="stop_id")]
        public int? StopId { get; set; }

        /// <summary>
        /// Transport mode identifier
        /// </summary>
        /// <value>Transport mode identifier</value>

        [DataMember(Name="route_type")]
        public int? RouteType { get; set; }

        /// <summary>
        /// Geographic coordinate of latitude at stop
        /// </summary>
        /// <value>Geographic coordinate of latitude at stop</value>

        [DataMember(Name="stop_latitude")]
        public float? StopLatitude { get; set; }

        /// <summary>
        /// Geographic coordinate of longitude at stop
        /// </summary>
        /// <value>Geographic coordinate of longitude at stop</value>

        [DataMember(Name="stop_longitude")]
        public float? StopLongitude { get; set; }

        /// <summary>
        /// Landmark in proximity of stop
        /// </summary>
        /// <value>Landmark in proximity of stop</value>

        [DataMember(Name="stop_landmark")]
        public string StopLandmark { get; set; }

        /// <summary>
        /// Sequence of the stop on the route/run; return 0 when route_id or run_id not specified. Order ascendingly by this field (when non zero) to get physical order (earliest first) of stops on the route_id/run_id.
        /// </summary>
        /// <value>Sequence of the stop on the route/run; return 0 when route_id or run_id not specified. Order ascendingly by this field (when non zero) to get physical order (earliest first) of stops on the route_id/run_id.</value>

        [DataMember(Name="stop_sequence")]
        public int? StopSequence { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class V3StoppingPatternStop {\n");
            sb.Append("  StopTicket: ").Append(StopTicket).Append("\n");
            sb.Append("  StopDistance: ").Append(StopDistance).Append("\n");
            sb.Append("  StopSuburb: ").Append(StopSuburb).Append("\n");
            sb.Append("  StopName: ").Append(StopName).Append("\n");
            sb.Append("  StopId: ").Append(StopId).Append("\n");
            sb.Append("  RouteType: ").Append(RouteType).Append("\n");
            sb.Append("  StopLatitude: ").Append(StopLatitude).Append("\n");
            sb.Append("  StopLongitude: ").Append(StopLongitude).Append("\n");
            sb.Append("  StopLandmark: ").Append(StopLandmark).Append("\n");
            sb.Append("  StopSequence: ").Append(StopSequence).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            return obj.GetType() == GetType() && Equals((V3StoppingPatternStop)obj);
        }

        /// <summary>
        /// Returns true if V3StoppingPatternStop instances are equal
        /// </summary>
        /// <param name="other">Instance of V3StoppingPatternStop to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(V3StoppingPatternStop other)
        {
            if (ReferenceEquals(null, other)) return false;
            if (ReferenceEquals(this, other)) return true;

            return 
                (
                    StopTicket == other.StopTicket ||
                    StopTicket != null &&
                    StopTicket.Equals(other.StopTicket)
                ) && 
                (
                    StopDistance == other.StopDistance ||
                    StopDistance != null &&
                    StopDistance.Equals(other.StopDistance)
                ) && 
                (
                    StopSuburb == other.StopSuburb ||
                    StopSuburb != null &&
                    StopSuburb.Equals(other.StopSuburb)
                ) && 
                (
                    StopName == other.StopName ||
                    StopName != null &&
                    StopName.Equals(other.StopName)
                ) && 
                (
                    StopId == other.StopId ||
                    StopId != null &&
                    StopId.Equals(other.StopId)
                ) && 
                (
                    RouteType == other.RouteType ||
                    RouteType != null &&
                    RouteType.Equals(other.RouteType)
                ) && 
                (
                    StopLatitude == other.StopLatitude ||
                    StopLatitude != null &&
                    StopLatitude.Equals(other.StopLatitude)
                ) && 
                (
                    StopLongitude == other.StopLongitude ||
                    StopLongitude != null &&
                    StopLongitude.Equals(other.StopLongitude)
                ) && 
                (
                    StopLandmark == other.StopLandmark ||
                    StopLandmark != null &&
                    StopLandmark.Equals(other.StopLandmark)
                ) && 
                (
                    StopSequence == other.StopSequence ||
                    StopSequence != null &&
                    StopSequence.Equals(other.StopSequence)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                var hashCode = 41;
                // Suitable nullity checks etc, of course :)
                    if (StopTicket != null)
                    hashCode = hashCode * 59 + StopTicket.GetHashCode();
                    if (StopDistance != null)
                    hashCode = hashCode * 59 + StopDistance.GetHashCode();
                    if (StopSuburb != null)
                    hashCode = hashCode * 59 + StopSuburb.GetHashCode();
                    if (StopName != null)
                    hashCode = hashCode * 59 + StopName.GetHashCode();
                    if (StopId != null)
                    hashCode = hashCode * 59 + StopId.GetHashCode();
                    if (RouteType != null)
                    hashCode = hashCode * 59 + RouteType.GetHashCode();
                    if (StopLatitude != null)
                    hashCode = hashCode * 59 + StopLatitude.GetHashCode();
                    if (StopLongitude != null)
                    hashCode = hashCode * 59 + StopLongitude.GetHashCode();
                    if (StopLandmark != null)
                    hashCode = hashCode * 59 + StopLandmark.GetHashCode();
                    if (StopSequence != null)
                    hashCode = hashCode * 59 + StopSequence.GetHashCode();
                return hashCode;
            }
        }

        #region Operators
        #pragma warning disable 1591

        public static bool operator ==(V3StoppingPatternStop left, V3StoppingPatternStop right)
        {
            return Equals(left, right);
        }

        public static bool operator !=(V3StoppingPatternStop left, V3StoppingPatternStop right)
        {
            return !Equals(left, right);
        }

        #pragma warning restore 1591
        #endregion Operators
    }
}

File: r/R/V3StoppingPatternStop.r
# PTV Timetable API - Version 3
#
# The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
#
# OpenAPI spec version: v3
# 
# Generated by: https://github.com/swagger-api/swagger-codegen.git

#' V3StoppingPatternStop Class
#'
#' @field stop_ticket 
#' @field stop_distance 
#' @field stop_suburb 
#' @field stop_name 
#' @field stop_id 
#' @field route_type 
#' @field stop_latitude 
#' @field stop_longitude 
#' @field stop_landmark 
#' @field stop_sequence 
#'
#' @importFrom R6 R6Class
#' @importFrom jsonlite fromJSON toJSON
#' @export
V3StoppingPatternStop <- R6::R6Class(
  'V3StoppingPatternStop',
  public = list(
    `stop_ticket` = NULL,
    `stop_distance` = NULL,
    `stop_suburb` = NULL,
    `stop_name` = NULL,
    `stop_id` = NULL,
    `route_type` = NULL,
    `stop_latitude` = NULL,
    `stop_longitude` = NULL,
    `stop_landmark` = NULL,
    `stop_sequence` = NULL,
    initialize = function(`stop_ticket`, `stop_distance`, `stop_suburb`, `stop_name`, `stop_id`, `route_type`, `stop_latitude`, `stop_longitude`, `stop_landmark`, `stop_sequence`){
      if (!missing(`stop_ticket`)) {
        stopifnot(R6::is.R6(`stop_ticket`))
        self$`stop_ticket` <- `stop_ticket`
      }
      if (!missing(`stop_distance`)) {
        stopifnot(is.numeric(`stop_distance`), length(`stop_distance`) == 1)
        self$`stop_distance` <- `stop_distance`
      }
      if (!missing(`stop_suburb`)) {
        stopifnot(is.character(`stop_suburb`), length(`stop_suburb`) == 1)
        self$`stop_suburb` <- `stop_suburb`
      }
      if (!missing(`stop_name`)) {
        stopifnot(is.character(`stop_name`), length(`stop_name`) == 1)
        self$`stop_name` <- `stop_name`
      }
      if (!missing(`stop_id`)) {
        stopifnot(is.numeric(`stop_id`), length(`stop_id`) == 1)
        self$`stop_id` <- `stop_id`
      }
      if (!missing(`route_type`)) {
        stopifnot(is.numeric(`route_type`), length(`route_type`) == 1)
        self$`route_type` <- `route_type`
      }
      if (!missing(`stop_latitude`)) {
        stopifnot(is.numeric(`stop_latitude`), length(`stop_latitude`) == 1)
        self$`stop_latitude` <- `stop_latitude`
      }
      if (!missing(`stop_longitude`)) {
        stopifnot(is.numeric(`stop_longitude`), length(`stop_longitude`) == 1)
        self$`stop_longitude` <- `stop_longitude`
      }
      if (!missing(`stop_landmark`)) {
        stopifnot(is.character(`stop_landmark`), length(`stop_landmark`) == 1)
        self$`stop_landmark` <- `stop_landmark`
      }
      if (!missing(`stop_sequence`)) {
        stopifnot(is.numeric(`stop_sequence`), length(`stop_sequence`) == 1)
        self$`stop_sequence` <- `stop_sequence`
      }
    },
    toJSON = function() {
      V3StoppingPatternStopObject <- list()
      if (!is.null(self$`stop_ticket`)) {
        V3StoppingPatternStopObject[['stop_ticket']] <- self$`stop_ticket`$toJSON()
      }
      if (!is.null(self$`stop_distance`)) {
        V3StoppingPatternStopObject[['stop_distance']] <- self$`stop_distance`
      }
      if (!is.null(self$`stop_suburb`)) {
        V3StoppingPatternStopObject[['stop_suburb']] <- self$`stop_suburb`
      }
      if (!is.null(self$`stop_name`)) {
        V3StoppingPatternStopObject[['stop_name']] <- self$`stop_name`
      }
      if (!is.null(self$`stop_id`)) {
        V3StoppingPatternStopObject[['stop_id']] <- self$`stop_id`
      }
      if (!is.null(self$`route_type`)) {
        V3StoppingPatternStopObject[['route_type']] <- self$`route_type`
      }
      if (!is.null(self$`stop_latitude`)) {
        V3StoppingPatternStopObject[['stop_latitude']] <- self$`stop_latitude`
      }
      if (!is.null(self$`stop_longitude`)) {
        V3StoppingPatternStopObject[['stop_longitude']] <- self$`stop_longitude`
      }
      if (!is.null(self$`stop_landmark`)) {
        V3StoppingPatternStopObject[['stop_landmark']] <- self$`stop_landmark`
      }
      if (!is.null(self$`stop_sequence`)) {
        V3StoppingPatternStopObject[['stop_sequence']] <- self$`stop_sequence`
      }

      V3StoppingPatternStopObject
    },
    fromJSON = function(V3StoppingPatternStopJson) {
      V3StoppingPatternStopObject <- jsonlite::fromJSON(V3StoppingPatternStopJson)
      if (!is.null(V3StoppingPatternStopObject$`stop_ticket`)) {
        stop_ticketObject <- V3StopTicket$new()
        stop_ticketObject$fromJSON(jsonlite::toJSON(V3StoppingPatternStopObject$stop_ticket, auto_unbox = TRUE))
        self$`stop_ticket` <- stop_ticketObject
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_distance`)) {
        self$`stop_distance` <- V3StoppingPatternStopObject$`stop_distance`
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_suburb`)) {
        self$`stop_suburb` <- V3StoppingPatternStopObject$`stop_suburb`
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_name`)) {
        self$`stop_name` <- V3StoppingPatternStopObject$`stop_name`
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_id`)) {
        self$`stop_id` <- V3StoppingPatternStopObject$`stop_id`
      }
      if (!is.null(V3StoppingPatternStopObject$`route_type`)) {
        self$`route_type` <- V3StoppingPatternStopObject$`route_type`
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_latitude`)) {
        self$`stop_latitude` <- V3StoppingPatternStopObject$`stop_latitude`
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_longitude`)) {
        self$`stop_longitude` <- V3StoppingPatternStopObject$`stop_longitude`
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_landmark`)) {
        self$`stop_landmark` <- V3StoppingPatternStopObject$`stop_landmark`
      }
      if (!is.null(V3StoppingPatternStopObject$`stop_sequence`)) {
        self$`stop_sequence` <- V3StoppingPatternStopObject$`stop_sequence`
      }
    },
    toJSONString = function() {
       sprintf(
        '{
           "stop_ticket": %s,
           "stop_distance": %d,
           "stop_suburb": %s,
           "stop_name": %s,
           "stop_id": %d,
           "route_type": %d,
           "stop_latitude": %d,
           "stop_longitude": %d,
           "stop_landmark": %s,
           "stop_sequence": %d
        }',
        self$`stop_ticket`$toJSON(),
        self$`stop_distance`,
        self$`stop_suburb`,
        self$`stop_name`,
        self$`stop_id`,
        self$`route_type`,
        self$`stop_latitude`,
        self$`stop_longitude`,
        self$`stop_landmark`,
        self$`stop_sequence`
      )
    },
    fromJSONString = function(V3StoppingPatternStopJson) {
      V3StoppingPatternStopObject <- jsonlite::fromJSON(V3StoppingPatternStopJson)
      V3StopTicketObject <- V3StopTicket$new()
      self$`stop_ticket` <- V3StopTicketObject$fromJSON(jsonlite::toJSON(V3StoppingPatternStopObject$stop_ticket, auto_unbox = TRUE))
      self$`stop_distance` <- V3StoppingPatternStopObject$`stop_distance`
      self$`stop_suburb` <- V3StoppingPatternStopObject$`stop_suburb`
      self$`stop_name` <- V3StoppingPatternStopObject$`stop_name`
      self$`stop_id` <- V3StoppingPatternStopObject$`stop_id`
      self$`route_type` <- V3StoppingPatternStopObject$`route_type`
      self$`stop_latitude` <- V3StoppingPatternStopObject$`stop_latitude`
      self$`stop_longitude` <- V3StoppingPatternStopObject$`stop_longitude`
      self$`stop_landmark` <- V3StoppingPatternStopObject$`stop_landmark`
      self$`stop_sequence` <- V3StoppingPatternStopObject$`stop_sequence`
    }
  )
)
